### ISSUE TITLE
[BUG] uploads folder points to sillytavern source, not dataRoot?

### ISSUE THREAD

**Issue by vikanezrimaya** [SillyTavern/SillyTavern#2422]
### Environment

🐧 Linux

### System

NixOS (self-packaged)

### Version

1.12.1

### Desktop Information

 - Node ver: v20.12.2
 - Branch: release (should be reproducible on staging, as the code is the same)

### Describe the problem

Launching SillyTavern with its code read-only tries to create an `uploads` dir within the source code directory and predictably fails.

### Additional info

This bug is a follow-up to #2020.

```
Jun 26 01:25:09 rafflesia sillytavern[124778]: Node version: v20.12.2. Running in undefined environment.
Jun 26 01:25:09 rafflesia sillytavern[124778]: /nix/store/0yxgyds1v2sx9pv60cq1qq9fhdy0ynwn-sillytavern-1.12.1/lib/sillytavern/node_modules/mkdirp/index.js:93
Jun 26 01:25:09 rafflesia sillytavern[124778]:                     throw err0;
Jun 26 01:25:09 rafflesia sillytavern[124778]:                     ^
Jun 26 01:25:09 rafflesia sillytavern[124778]: Error: EROFS: read-only file system, mkdir '/nix/store/0yxgyds1v2sx9pv60cq1qq9fhdy0ynwn-sillytavern-1.12.1/lib/sillytavern/uploads'
Jun 26 01:25:09 rafflesia sillytavern[124778]:     at Object.mkdirSync (node:fs:1373:26)
Jun 26 01:25:09 rafflesia sillytavern[124778]:     at Function.sync (/nix/store/0yxgyds1v2sx9pv60cq1qq9fhdy0ynwn-sillytavern-1.12.1/lib/sillytavern/node_modules/mkdirp/index.js:74:13)
Jun 26 01:25:09 rafflesia sillytavern[124778]:     at new DiskStorage (/nix/store/0yxgyds1v2sx9pv60cq1qq9fhdy0ynwn-sillytavern-1.12.1/lib/sillytavern/node_modules/multer/storage/disk.js:21:12)
Jun 26 01:25:09 rafflesia sillytavern[124778]:     at module.exports (/nix/store/0yxgyds1v2sx9pv60cq1qq9fhdy0ynwn-sillytavern-1.12.1/lib/sillytavern/node_modules/multer/storage/disk.js:65:10)
Jun 26 01:25:09 rafflesia sillytavern[124778]:     at new Multer (/nix/store/0yxgyds1v2sx9pv60cq1qq9fhdy0ynwn-sillytavern-1.12.1/lib/sillytavern/node_modules/multer/index.js:15:20)
Jun 26 01:25:09 rafflesia sillytavern[124778]:     at multer (/nix/store/0yxgyds1v2sx9pv60cq1qq9fhdy0ynwn-sillytavern-1.12.1/lib/sillytavern/node_modules/multer/index.js:95:12)
Jun 26 01:25:09 rafflesia sillytavern[124778]:     at Object.<anonymous> (/nix/store/0yxgyds1v2sx9pv60cq1qq9fhdy0ynwn-sillytavern-1.12.1/lib/sillytavern/server.js:289:9)
Jun 26 01:25:09 rafflesia sillytavern[124778]:     at Module._compile (node:internal/modules/cjs/loader:1369:14)
Jun 26 01:25:09 rafflesia sillytavern[124778]:     at Module._extensions..js (node:internal/modules/cjs/loader:1427:10)
Jun 26 01:25:09 rafflesia sillytavern[124778]:     at Module.load (node:internal/modules/cjs/loader:1206:32) {
Jun 26 01:25:09 rafflesia sillytavern[124778]:   errno: -30,
Jun 26 01:25:09 rafflesia sillytavern[124778]:   code: 'EROFS',
Jun 26 01:25:09 rafflesia sillytavern[124778]:   syscall: 'mkdir',
Jun 26 01:25:09 rafflesia sillytavern[124778]:   path: '/nix/store/0yxgyds1v2sx9pv60cq1qq9fhdy0ynwn-sillytavern-1.12.1/lib/sillytavern/uploads'
Jun 26 01:25:09 rafflesia sillytavern[124778]: }
Jun 26 01:25:09 rafflesia sillytavern[124778]: Node.js v20.12.2
```

Line triggering the bug: [server.js:289](https://github.com/SillyTavern/SillyTavern/blob/release/server.js#L289) — `UPLOADS_PATH` is pulled from constants and is relative to current directory, which is the directory where SillyTavern hosts its code. It should instead be relative to `dataRoot`. ~~Presumably whatever `multer` is should handle exposing this folder to HTTP anyway.~~ Additional code changes will be required to accommodate this.

### Please tick the boxes

- [X] I have explained the issue clearly, and I included all relevant info
- [X] I have checked that this [issue hasn't already been raised](https://github.com/SillyTavern/SillyTavern/issues?q=is%3Aissue)
- [X] I have checked the [docs](https://docs.sillytavern.app/) ![important](https://img.shields.io/badge/Important!-F6094E)

**Comment by Cohee1207** [comment]
Fixed in #2426

**Comment by Cohee1207** [comment]
I moved uploads only.
cache (for transformers models) and access.log (for web server logs) still use relative paths to the server.

**Comment by vikanezrimaya** [comment]
> cache (for transformers models) and access.log (for web server logs) still use relative paths to the server.

Those *probably* should also get moved to the data root (or have different, configurable paths — I put data in `/var/lib/sillytavern`, so I'd expect cache to go to `/var/cache/sillytavern` and logs to `/var/log/sillytavern`) but at least those don't make the entire software crash on start, so that can be fixed later, I guess.

I looked through the patch and found it much higher in quality than the crude patch I wrote to work around this issue. I'll probably replace the patch I locally made with this one. Thank you for this! <3
